src/components/ProgramForm/ThreeColProgram/List/Recommender/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { getCourseRecommender, getExerciseRecommender } from '@/services/escola-lms/recommender';
+import { FormattedMessage, history } from 'umi';
+import { Context } from '../../../Context';
+import { RecommenderType, TopicType } from '@/services/escola-lms/enums';
+
+import { ExerciseRecommender } from './ExerciseRecommender';
+import { CourseRecommender } from './CourseRecommender';
+import { RecommenderIcon } from '@/icons';
+import { RecommenderInfo } from './RecommenderInfo';
+import { Spin } from 'antd';
+
+export const Recommender: React.FC<{
+  courseId: number;
+  lessonId: number;
+  recommenderType?: RecommenderType;
+}> = ({ courseId, lessonId, recommenderType = RecommenderType.Exercise }) => {
+  const [course, setCourse] = useState<API.Recommender>();
+  const [excercise, setExcercise] = useState<API.RecommenderExercise>();
+  const { addNewTopic } = useContext(Context);
+
+  const onTopicCreate = useCallback(
+    (lesson_id: number, topic_type: TopicType) => {
+      if (addNewTopic) {
+        const newTopic = addNewTopic(lesson_id, topic_type);
+        history.push(`/courses/list/${courseId}/program/?topic=${newTopic.id}`);
+      }
+    },
+    [addNewTopic],
+  );
+
+  const getTopicType = (topic: API.RecommenderTopicType) => (topic.includes('H5P') ? 'H5P' : topic);
```

Comment:
This should be outside of component scope - it's pure function. I would also add return type.